packages/@aws-cdk/aws-codedeploy/lambda-packages/ecs-deployment-provider/lib/is-complete.ts

```diff
@@ -0,0 +1,128 @@
+            if (rollbackResp.deploymentInfo?.status) {
+              if (rollbackResp.deploymentInfo?.status == DeploymentStatus.SUCCEEDED ||
+                rollbackResp.deploymentInfo?.status == DeploymentStatus.FAILED ||
+                rollbackResp.deploymentInfo?.status == DeploymentStatus.STOPPED) {
+                const errInfo = resp.deploymentInfo.errorInformation;
+                const error = new Error(`Deployment ${resp.deploymentInfo.status}: [${errInfo?.code}] ${errInfo?.message}`);
+                logger.error('Deployment failed', { complete: true, error });
+                throw error;
+              }
+              logger.info('Waiting for final status from a rollback', { complete: false });
+
+              return { IsComplete: false }; // waiting for final status from rollback
+            } else {
+              const errInfo = resp.deploymentInfo.errorInformation;
+              const error = new Error(`Deployment ${resp.deploymentInfo.status}: [${errInfo?.code}] ${errInfo?.message}`);
+              logger.error('No rollback to wait for', { complete: true, error });
+              throw error;
+            }
+          default:
+            logger.info('Waiting for final status from deployment', { complete: false });
+
+            return { IsComplete: false };
+        }
+      case 'Delete':
+        switch (resp.deploymentInfo?.status) {
+          case DeploymentStatus.SUCCEEDED:
+            logger.info('Deployment finished successfully - nothing to delete', { complete: true });
+
+            return { IsComplete: true };
+          case DeploymentStatus.FAILED:
+          case DeploymentStatus.STOPPED:
+            if (rollbackResp.deploymentInfo?.status) {
+              if (rollbackResp.deploymentInfo?.status == DeploymentStatus.SUCCEEDED ||
+                rollbackResp.deploymentInfo?.status == DeploymentStatus.FAILED ||
+                rollbackResp.deploymentInfo?.status == DeploymentStatus.STOPPED) {
+                logger.info('Rollback in final status', { complete: true });
+
+                return { IsComplete: true }; // rollback finished, we're deleted
+              }
+              logger.info('Waiting for final status from a rollback', { complete: false });
+
+              return { IsComplete: false }; // waiting for rollback
+            }
+            logger.info('No rollback to wait for', { complete: true });
+
+            return { IsComplete: true };
+          default:
+            logger.info('Waiting for final status from deployment', { complete: false });
+
+            return { IsComplete: false };
+        }
+      default:
+        logger.error('Unknown request type');
+        throw new Error(`Unknown request type: ${event.RequestType}`);
+    }
+  } catch (e) {
+    logger.error('Unable to determine deployment status', e as Error);
+    if (event.RequestType === 'Delete') {
+      logger.warn('Ignoring error - nothing to do', { complete: true });
```

Comment:
I think we do want to fail under certain scenarios. If there is still a deployment in process there is some error deleting it, but we make this resource as successfully deleted we might run into errors trying to delete downstream resources (can you delete a deployment group while a deployment is in process?)

Reply:
I think this is asking for trouble. Raising errors from custom resources during delete events is the quickest way to get a stack into a state in which it can't be updated or rolled back cleanly and the only resort is to terminate and start over.